src/Process/Root.php

```diff
@@ -14,11 +22,35 @@ public function __construct()
 
     protected function _run(): Forked
     {
+        $this->_getProcessCollection()->applyProcessPool($this->_getProcessPool());
+
         while (true) {
             $this->getProcessSignalDispatcher()->processBufferedSignals();
+            $this->pollSingletonProcesses();
             sleep(1);
         }
 
         return $this;
     }
+
+    protected function pollSingletonProcesses() : Root
+    {
+        foreach (self::SINGLETON_PROCESSES as $singletonType) {
+            $semaphoreResource = $this->_getSemaphoreResource($singletonType);
+
+            // soft test the lock to spawn a process
+            // that process will go on to attempt to actually acquire the global mutex
+            if ($semaphoreResource->testLock()) {
+                try {
+                    $process = $this->_getProcessCollection()->getProcessPrototypeClone($singletonType);
+                    $this->_getProcessPool()->addChildProcess($process);
+                } catch (Forked\Exception $forkedException) {
+                    // this is fine, another execution environment will spawn this process
```

Comment:
```suggestion
                    // this is fine, another execution environment will spawn this process
                    $this->_getLogger()->debug($forkedException->getMessage(), ['exception' => $forkedException]);
```
We should make this event visible at least. We normal emit a `critical` level message when catching `Forked\Exception`s , but this is not `critical` yet based on your comment.